### DISCUSSION TITLE
Token timeline implementation: exploring endorsement chain vs token-registry-subgraph approach

### DISCUSSION THREAD

**Discussion by marlowl** [TradeTrust/tradetrust-website#803] [General]
We (Docklab) are searching for an effective approach to implement a token timeline in our frontend. I believe there are two potential methods to achieve this:

1.  The endorsement chain implementation approach used in this tradetrust-website repository.
2.  Utilizing the token-registry-subgraph approach developed here: [https://github.com/Open-Attestation/token-registry-subgraph](https://github.com/Open-Attestation/token-registry-subgraph).

@superical, do you think the subgraph approach is already mature enough for battle-testing, or would the current `EndorsementChain` approach be more suitable? 

Curious to hear opinions on this! :smile:

**Comment by superical** [comment]
heyy @marlowl, how've you been? It's been a while since your last post!😀

Both methods can retrieve info for the endorsement chain. If resources allow, I find it simpler to query against the subgraph for endorsement chains sometimes with fewer requests, which can be helpful if you expect to perform this frequently. Moreover, the token registry subgraph offers versatility for building other features too. Having said that, the approach used in the tradetrust-website also works for building the endorsement chain, especially if the overhead of using subgraph poses a challenge.

**Comment by marlowl** [comment]
@superical Haha, it has indeed been a while. All good with me, hope you are too! 😄 

By now, we have implemented the complete TradeTrust token-registry v4 token calls, verification api and custom document renderer  in our frontend, and we are still exploring improvement features like the token timeline / endorsement chain.

I can imagine that querying against a subgraph will be easier. On our end, we'll have to dive deeper and learn about the Graph protocol and subgraphs, but I believe we can make room for that. 

Would you, by any chance, be interested in a Docklab/TradeTrust tech team intro meeting?
I believe it would be beneficial for introductions and sharing updates on our current projects, as an alternative to using Github Discussions.

**Reply by superical** [reply]
Feel free to drop me a message. Github is still our preferred channel for issues or discussions related to TradeTrust though because there are areas some of our team members are more familiar with and could provide better help.🤗

**Comment by superical** [comment]
@marlowl I believe you already have my contacts if you’d like to reach me. I will close this thread for now. Feel free to reopen it if needed.🙂

**Reply by marlowl** [reply]
@superical sended you a proton mail! 😄

**Reply by superical** [reply]
Cool! I've been bogged down in some other work recently. I'll touch base with you very soon.🙇‍♂️

**Reply by marlowl** [reply]
@superical 👍🏻

**Reply by marlowl** [reply]
@superical please let me know when you are available to respond to the email! 😄

**Reply by marlowl** [reply]
@superical thanks for the response, responded back 👍🏻
